python/ray/util/rpdb.py

```diff
@@ -308,27 +308,27 @@ def _driver_set_trace():
     This disables Ray driver logs temporarily so that the PDB console is not
     spammed: https://github.com/ray-project/ray/issues/18172
     """
-    if ray.util.ray_debugpy._is_ray_debugger_enabled():
-        return ray.util.ray_debugpy.set_trace()
-
-    print("*** Temporarily disabling Ray worker logs ***")
-    ray._private.worker._worker_logs_enabled = False
+    if os.environ.get("RAY_DEBUG", "1") == "1":
+        ray.util.ray_debugpy.set_trace()
+    if os.environ.get("RAY_DEBUG", "1") == "legacy":
+        print("*** Temporarily disabling Ray worker logs ***")
+        ray._private.worker._worker_logs_enabled = False
 
-    def enable_logging():
-        print("*** Re-enabling Ray worker logs ***")
-        ray._private.worker._worker_logs_enabled = True
+        def enable_logging():
+            print("*** Re-enabling Ray worker logs ***")
+            ray._private.worker._worker_logs_enabled = True
 
-    pdb = _PdbWrap(enable_logging)
-    frame = sys._getframe().f_back
-    pdb.set_trace(frame)
+        pdb = _PdbWrap(enable_logging)
+        frame = sys._getframe().f_back
+        pdb.set_trace(frame)
 
 
-def _is_ray_debugger_enabled():
-    return "RAY_PDB" in os.environ or ray.util.ray_debugpy._is_ray_debugger_enabled()
+def _is_ray_debugger_post_mortem_enabled():
+    return os.environ.get("RAY_DEBUG_POST_MORTEM", "0") == "1"
 
 
 def _post_mortem():
-    if ray.util.ray_debugpy._is_ray_debugger_enabled():
+    if os.environ.get("RAY_DEBUG", "1") == "1":
```

Comment:
Just to confirm, we still want to enable post-mortem debugging when `RAY_DEBUG=0` & `RAY_DEBUG_POST_MORTEM=1`, right? 

If not, then I think we need to check `RAY_DEBUG=0` and do no-op here.

Reply:
Good point, thanks for pointing this out. I feel like if somebody explicitly opts into post mortem debugging, they probably want it, but if people think it is not intuitive we can change it :)